test/functional/test_framework/test_framework.py

```diff
@@ -194,6 +207,8 @@ def parse_args(self, test_file):
                             help="use BIP324 v2 connections between all nodes by default")
         parser.add_argument("--v1transport", dest="v1transport", default=False, action="store_true",
                             help="Explicitly use v1 transport (can be used to overwrite global --v2transport option)")
+        parser.add_argument("--test_methods", dest="test_methods", nargs='*',
+                            help="Run specified test methods sequentially instead of the full test. Use only for methods that do not depend on any context set up in run_test or other methods.")
```

Comment:
The PR title states that this will enable running functional test sub-tests but any method can be run with this argument at the moment such as one below. Should we restrict only for those that have the `test_` prefix?

```
+    def sample_calculator(self):
+        self.log.info("Sample calculator running")
+        x = 2 + 3
+        self.log.info(f'Calculated x to be: {x}')
+
```

Reply:
Some tests might not have the test prefix, though. I don’t think we can use the `test_` prefix to catch all sub-tests; rather, the goal is just to enable running individual methods that can be sub-tests.

Reply:
> or other methods.

Nit: This can be reworded though because the following snippet works.

```
    def calculate(self, val):
        return val

    def sample_calculator(self):
        self.log.info("Sample calculator running")
        x = self.calculate(2 + 3)
        self.log.info(f'Calculated x to be: {x}')
```

```
➜  bitcoin git:(09-2024-run-ind-functional-test) ✗ build/test/functional/feature_config_args.py --test_methods test_log_buffer test_args_log sample_calculator
```

Reply:
leaving this as is; but happy to take explicit suggestions :)

Reply:
If I'm not missing something, why not just remove it?

Reply:
It is useful to because running `build/test/functional/test_runner.py --help`, will describe this new feature for users. If this is removed, it will not be documented, and future users may not learn about it.
Therefore, I will be leaving it as is.

Reply:
I meant removing only "or other methods".

Reply:
Happy to update when retouching.